### PR TITLE
Refactor: Rename AWSElastic to AutoDiscovery for AWS and GCP compatibility

### PR DESCRIPTION
**Pull request by hazdikk**
This PR includes a refactor of the codebase to improve compatibility between AWS and GCP Auto Discovery. Previously, the code was specifically named and designed for AWS ElasticCache. However, after careful review and testing, it was found that AWS ElasticCache and GCP Auto Discovery work in the same way. 

The commands and responses for both services are identical, which means we can use the same methods for both. This refactor renames all instances of `AWSElastic` to `AutoDiscovery` to reflect this compatibility and to avoid confusion for users. 

Users can now use the same `AutoDiscovery` methods for both AWS and GCP Auto Discovery. 

For more information on how to use AWS Auto Discovery, please refer to the [AWS Documentation](https://docs.aws.amazon.com/AmazonElastiCache/latest/mem-ug/AutoDiscovery.AddingToYourClientLibrary.html). 

For more information on how to use GCP Auto Discovery, please refer to the [GCP Documentation](https://cloud.google.com/memorystore/docs/memcached/use-auto-discovery).

Please note that this refactor does not change the functionality of the methods, but simply renames them for clarity and improved compatibility.

**Note:** The GitHub Wiki has not been updated to reflect these changes as I am currently unsure of how to do so. Any assistance or guidance on this would be appreciated.

### CODE REVIEW

**Comment by killme2008** [comment]
It's a good idea. But i don't like the breaking change, can we provide some way to keep compatible with previous versions, and make the `AWSElastic` deprecated?

**Comment by hazdikk** [comment]
Sure, will update later, thank for the input

**Comment by hazdikk** [comment]
Updated

File: src/main/java/net/rubyeye/xmemcached/aws/AWSElasticCacheClient.java [hunk 1]
```diff
@@ -38,6 +42,7 @@
  * @author dennis
  *
```
Comment (killme2008) [inline src/main/java/net/rubyeye/xmemcached/aws/AWSElasticCacheClient.java:43]: Adds a link to `AutoDiscoveryCacheClient`.

```suggestion
 * @see  net.rubyeye.xmemcached.autodiscovery.AutoDiscoveryCacheClient
 *
```

File: src/main/java/net/rubyeye/xmemcached/aws/AWSElasticCacheClientBuilder.java [hunk 2]
```diff
@@ -13,6 +13,7 @@
  * @author dennis
  *
```
Comment (killme2008) [inline src/main/java/net/rubyeye/xmemcached/aws/AWSElasticCacheClientBuilder.java:14]:
```suggestion
 * @see  net.rubyeye.xmemcached.autodiscovery.AutoDiscoveryCacheClientBuilder
 *
```

**Review by killme2008 — APPROVED** [review]
Almost LGTM

**Pull request merged**